apps/web/src/app/[lang]/[...path]/page.tsx

```diff
@@ -27,7 +27,34 @@ const getPage = async (path: string[], lang: Locale) => {
     locale: lang,
   });
 
-  if (!page) return notFound();
+  if (!page) {
+    const otherLang = lang === "fi" ? "en" : "fi";
+    const localizedSlug = `slug.${otherLang}` as const;
+    const localizedTopic = `topic.${otherLang}` as const;
+    const otherPage = await fetchPage({
+      // @ts-expect-error Typescript doesn't get as const keys in object assignments it seems
+      where:
+        path.length === 1
+          ? { [localizedSlug]: { equals: path[0] } }
+          : {
+              [localizedSlug]: { equals: path[1] },
+              [localizedTopic]: { [localizedSlug]: { equals: path[0] } },
+            },
+      locale: "all",
```

Comment:
If possible, we should try and avoid this at all costs, OR use this always and for everything, it's either or really. This will break caching currently if done this way, but we could look into just doing caching for the "all" locale endpoints always, would perhaps simplify stuff.

Reply:
It would be a bit scuffed to traverse the responses to extract the given locale data for "all" locale requests. And there is no type support from payload end. I'll try to figure out how to do caching here, but probably makes sense to always invalidate the "all" tag if any language version of a page is changed.

Reply:
Now it looks like it's working :sweat_smile: